### ISSUE TITLE
Kitkat not supported?

### ISSUE THREAD

**Issue by anonymous73** [bitcoin-wallet/bitcoin-wallet#574]
Hello,  I've gone to some lengths to ensure that my new KitKat installation/firmware is as bug, trojan and Google-free as possible but today I learn, from the latest changelog, that the app will no longer be supported on Kitkat and Lollipop will be the earliest version of Android supported.

I think this is a mistake. Internationally people can still buy Kitkat Android phones, especially people in countries that would benefit hugely from the freedom the bitcoin protocol provides. 

Therefore I sincerely ask that you reconsider dropping Kitkat (or any earlier versions that can still be bought provided that they do not have any gaping vulnerabilities).

Bitcoin needs to be open to as many people as possible.

On a related note, F-Droid says that I can update to one of the 6.41 versions in the list so I'm confused as to whether I can update or not (or should).

The changelog says that phase one of segwit is implemented in this release. This is exciting from a scalability perspective. I must admit though,  I have yet to find an explanation of segwit that satisfies me. So as a side note if someone finds a good technical explanation beyond the mere "fast lane" explanation that would be great. 

Thanks and please get back to me on this

**Comment by schildbach** [comment]
v6.41 is the currently last version that still supports KitKat (Android 4.4). It's still supported with security updates and important bugfixes for a while.

Can you give an example for a popular KitKat-only phone that is still produced and sold? What phone model are you using and how do you make sure all known vulnerablilities are fixed?

Segwit is not particularly exciting from a user perspective. It fixes a few known problems, it's a _small_ optimization in terms of blockchain capacity (transactions/second) and it's a preparation for easier future enhancements.

**Comment by anonymous73** [comment]
Thanks for responding. Good to hear that it is still being supported minimally,  but if possible, I do ask that you try to continue to fully maintain Kitkat.

Basically vendors still sell Samsungs that use KitKat, and they are really great small phones. The firmware that arrives on the phones is old and has vulnerabilities but with a bit of work I have upgraded to a firmware that fixes the known vulnerability that I encountered. Since then I have not had any *identifiable* attacks on this phone, I have gone to a few lengths to try to reduce the attack surface too like disabling bloat etc. That being said there may be vulnerabilities, but any setup cannot be guaranteed 100% impervious. Especially ones connected to the web so I do take extra care.

Thanks for your words on segwit. So segwit in itself is not a big deal? I thought segwit and lightning network were the same thing,  are there any good resources on these aspects?

**Comment by schildbach** [comment]
A quick analysis of the [Android Security Bulletins](https://source.android.com/security/bulletin) yields that Google stopped backporting security fixes for Android 4.4 with security patch level 2017-10-01. I strongly doubt any of the KitKat devices have received a significant number of security fixes since then. Even before, some vendors were too lazy to `git cherry-pick` and simply increased the patch level, leaving users at risk.

**I think it's more than time to leave Android 4.4 behind for security sensitive use cases.**

On top of that, I'd like to progress the UI/UX which was held back until recently by the requirement to support KitKat.

Lightning network has nothing to do with segwit, except for the fact it needs one of the segwit fixes to run safely.

**Comment by anonymous73** [comment]
Thanks for the reference.

I will do more research into how I can safely upgrade my firmware to Lollipop with my hardware!

**Comment by schildbach** [comment]
If you can, I suggest you try to upgrade to Android 7.0 (Nougat) or later. Google already dropped backports to all older releases since security patch level 2018-09-01.

**Comment by anonymous73** [comment]
Bloody Google. There's a reason why they dropped, "Don't Be Evil". Forced obsolesence of this magnitude is disgusting.

**Comment by anonymous73** [comment]
Hello Dear Schilbach,
Sadly I am getting conflicting reports are to whether this device can even (officially) handle 5.0, due to RAM related concerns by particular carriers. Ideally, I would upgrade to 5.2 which has a dark theme to save battery on the AMOLED display. Looks like I need to buy a new phone with silly things that I do not want like, batteries that you cannot change, and fingerprint sensors, lower resolution etc. Sad.

Do you think that very important updates will not be done, if they are extremely urgent? For example when a really bad low level exploit was found in the way certain messages were received this was fixed and a bunch of other (less urgent) patches fixed at the same time etc.  This suggests to me that now all updates are of extreme importance and if a vulnerability were to be found that puts people at high risk then then developers begrudgingly update it. I could be wrong.

If you are concerned about the look-and-feel being "held back" I have been pleasantly surprised by the look-and-feel of Open Keychain (F-Droid), Etar (Calender app on F-Droid) and Tasks (F-Driod, Turquiose coloured icon). Is there something specific that the new phones can do graphically? I admit that I'm not an expert, so maybe their GPUs do some cool graphical stuff that you are excited to implement.

I don't know if these few extra things are something worth considering.

SIDENOTE (As I said earlier if anyone can point me to where I can learn more about segwit and lightning network details, I want to learn about it. This will arm me in my travels when talking to people about bitcoin too. Knowledge is power and I want bitcoin to have lots of power ;)

**Comment by anonymous73** [comment]
CORRECTION TO 2nd PARAGRAPH: "This suggests to me that **not** all updates are of extreme importance..."

**Comment by anonymous73** [comment]
Hello Dear @schildbach. I'm not sure if you have had a chance to see the above, or what your thoughts are on the above.

 I do have design skills and would be happy to help out in future, especially if you have cosmetic changes planned, but I'm thinking that it might be a mistake to drop Kitkat given that critical updates appear to be done.

**Comment by schildbach** [comment]
As I said, the 6.x version is still supported with security updates and critical bugfixes for a while. No, it won't get any UI updates and cosmetic changes.

**Comment by anonymous73** [comment]
Understood. However if there are no critical updates that have gone unpatched I think it would be a mistake. 

BTW, the way you answered my offer to help with enhancing the look-and-feel, makes me think there is a misunderstanding. So there is no mistake, I will clarify. I was not asking for any cosmetic changes. In an earlier response you stated that you felt the program was being "held back" due to its support for KitKat. I responded above that there are some great, modern, slick apps for KitKat on F-Droid, like Etar, Open Keychain and others. In my most recent comment I simply offered to volunteer my services as a graphic designer, (and web developer) to aid you in any future work. My offer still stands.

My only concern is that support is not pulled too prematurely from KitKat.

Quick questions (for anyone). 1. Are there any bounties on a KitKat device?

Q2: As asked previosly,  does anyone know of a good explanation of segwit and lightning network for educational purposes? Too many people criticise Bitcoin's scalability, so a good explanation *for someone who already* understands asymmetric cryptography, and how the blockchain solves the double-spend problem. Basically I have yet to come across a good explanation and I did try searching.

By for now.
